src/ireiat/data_pipeline/assets/demand/faf5_tons.py

```diff
@@ -11,54 +11,155 @@
+        [FAF_TONS_TARGET_FIELD]
+    ].sum()
+
+    # Pivot by origin/destination to calculate percentages for each mode
+    mode_pivot_pdf = known_agg_pdf.pivot_table(
+        index=["dms_orig", "dms_dest"],
+        columns="dms_mode",
+        values=FAF_TONS_TARGET_FIELD,
+        fill_value=0,
+    )
+
+    # Add total tons and calculate percentages
+    mode_pivot_pdf["total_tons"] = mode_pivot_pdf.sum(axis=1)
+    mode_pivot_pdf["truck_pct"] = mode_pivot_pdf[FAFMode.TRUCK] / mode_pivot_pdf["total_tons"]
+    mode_pivot_pdf["rail_pct"] = mode_pivot_pdf[FAFMode.RAIL] / mode_pivot_pdf["total_tons"]
+    mode_pivot_pdf["water_pct"] = mode_pivot_pdf[FAFMode.WATER] / mode_pivot_pdf["total_tons"]
+
+    # Filter unknown origin/destination pairs that don't exist in known modes
+    valid_orig_dest_pairs = known_agg_pdf[["dms_orig", "dms_dest"]].drop_duplicates()
+
+    # Calculate unique origin-destination pairs in unknown modes before filtering
+    original_unknown_pairs = unknown_modes_pdf[["dms_orig", "dms_dest"]].drop_duplicates()
+
+    # Merge unknown modes with the known mode proportions, filter invalid pairs
+    unknown_modes_pdf = unknown_modes_pdf.merge(
+        mode_pivot_pdf[["truck_pct", "rail_pct", "water_pct"]],
+        on=["dms_orig", "dms_dest"],
+        how="left",
+    )
 
-    # Loop through each mode and generate the corresponding demand DataFrame
-    for mode in FAFDemandByMode:
-        mode_str = mode.name  # e.g., "TRUCK", "RAIL"
+    # Filter unknown origin/destination pairs that don't exist in known modes by merging with valid pairs
+    unknown_modes_pdf = unknown_modes_pdf.merge(
+        valid_orig_dest_pairs, on=["dms_orig", "dms_dest"], how="inner"
+    )
 
-        is_by_mode = faf_containerizable_pdf["dms_mode"] == FAFMode[mode_str]
-        faf_mode_pdf = faf_containerizable_pdf.loc[is_by_mode]
+    # Calculate the dropped pairs
+    remaining_unknown_pairs = unknown_modes_pdf[["dms_orig", "dms_dest"]].drop_duplicates()
+    dropped_pairs_count = len(original_unknown_pairs) - len(remaining_unknown_pairs)
 
-        # Group by origin and destination to calculate total tons
-        total_tons_od_pdf = faf_mode_pdf.groupby(["dms_orig", "dms_dest"], as_index=False)[
-            [FAF_TONS_TARGET_FIELD]
-        ].sum()
+    # Log a warning about the dropped origin-destination pairs
+    if dropped_pairs_count > 0:
+        context.log.warning(
+            f"{dropped_pairs_count} unique dms origin-destination pairs were dropped because they don't exist in the known (truck, rail, water) modes."
+        )
 
-        # Log and store the DataFrame, providing the correct output_name for multi-asset
-        context.log.info(f"{mode_str} mode demand generated with {len(total_tons_od_pdf)} records.")
-        publish_metadata(context, total_tons_od_pdf, output_name=mode.value)
+    # Explode unknown mode rows proportionally into truck, rail, water
```

Comment:
I'm just trying to understand what is going on here.  For a specific orig/dest pair, if there are, say 100 tons of unknown mode, 1000 tons of rail, 2000 tons of truck and 500 tons of water, so rail is 1000/3500 = 29%, truck is 57% and water is 14%, you will allocate that 100 tons of unknown to begin 29 tons of rail, 57 tons of truck and 14 tons of water?

Reply:
Yes. It's grossly wrong but at least it's O-D combo specific...and it gives us a starting point to actually be putting tons onto the network to solve the problem and we can return to refine.